### PR TITLE
Apply tags on updates in source-git

### PR DESCRIPTION
**Pull request by csomh**
In order to be able to track which dist-git commit did source-git update
originate from, create a "conversion" tag from the dist-git branch name
and commit hash (ex: `convert/c8s/05262ab9019701b9d278b305017861200204f559`)
and apply it at the tip of the branch in source-git.

Before triggering the conversion, check if such a tag exists, and skip
updating the source-git repo if it does.

### CODE REVIEW

File: deployment/roles/deploy/templates/worker-is.yml.j2 [hunk 1]
```diff
@@ -9,7 +9,7 @@ metadata:
   name: worker
 spec:
   tags:
-    - name: {{ deployment }}
+    - name: latest
```
Comment (csomh) [inline deployment/roles/deploy/templates/worker-is.yml.j2:12]: @jpopelka I've changed this back, to make the re-deployment work again when the image stream is updated.

I cannot think of a technical reason, why the image stream tag should match the deployment name. Am I missing something?

File: dist2src/core.py [hunk 2]
```diff
@@ -464,6 +464,34 @@ def perform_convert(self, origin_branch: str, dest_branch: str):
...
+                "\n"
+            )
+            if tag.startswith("imports/")
+        ]
+        if len(import_tags) > 1:
+            raise RuntimeError(
+                f"More than one import tag found on commit {dist_git_head}: {import_tags}"
+            )
+        if not import_tags:
+            raise RuntimeError(f"No import tags found on commit {dist_git_head}")
```
Comment (TomasTomecek) [inline dist2src/core.py:485]: I wonder what would be the resolution when this happens

**Review by TomasTomecek — COMMENTED** [review]
LGTM so far, I wonder though what to do about the two error cases - because they result in our source-git repo not being up to date.

File: deployment/roles/deploy/templates/worker-is.yml.j2 (hunk 1 above)
Comment (jpopelka) [reply deployment/roles/deploy/templates/worker-is.yml.j2:12]: Yes, should be OK.

File: dist2src/core.py (hunk 2 above)
Comment (csomh) [reply dist2src/core.py]: I've changed the method, as it turns out not all branches in dist-git have an import tag at the tip of the branch.

Instead we are creating our own tag using the name of the dist-git branch and the hash of the commit at the tip of that branch (ex: `convert/c8s/05262ab9019701b9d278b305017861200204f559`) and applying that one in source-git.

See and example [here](https://git.stg.centos.org/fork/csomh/source-git/acl/commits/c8).

**Comment by csomh** [comment]
recheck

**Comment by csomh** [comment]
@TomasTomecek please take another look. The `systemd` test that failed twice in a row here has passed for me locally, so if you agree, I would ignore the result of that here.

**Comment by csomh** [comment]
recheck

**Comment by csomh** [comment]
Okay, so the systemd test fails at `packit srpm`, with the message:

```
packit.exceptions.PackitSRPMException: Preparing of the upstream to the SRPM build failed: Failed to parse SPEC file: error: Unable to open /home/packit/rpmbuild/SOURCES/triggers.systemd: No such file or directory
```
Which does not make too much sense. Maybe this is a packit issue after all? :thinking:

**Comment by TomasTomecek** [comment]
Seems like systemd dist-git got updated and that update break our conversion process (did they remove the file?)

**Comment by csomh** [comment]
recheck

**Review by TomasTomecek — APPROVED** [review]

**Pull request merged**